lib/cucumber/core/report/summary.rb

```diff
@@ -4,20 +4,22 @@ module Report
       class Summary
         attr_reader :test_cases, :test_steps
 
-        def initialize
+        def initialize(event_bus)
           @test_cases = Test::Result::Summary.new
           @test_steps = Test::Result::Summary.new
+          subscribe_to(event_bus)
         end
 
-        def after_test_case(test_case, result)
-          result.describe_to test_cases
-        end
-
-        def after_test_step(test_step, result)
-          result.describe_to test_steps
-        end
+        private
 
-        def method_missing(*)
+        def subscribe_to(events)
```

Comment:
With the need to filter out the `steps` from the `hooks` (7bbae8b), it is not as straight forward to extract the `subscribe_to_event` method.